### ISSUE TITLE
How to detect the object of w/o gt

### ISSUE THREAD

**Issue by tsly123** [pierluigiferrari/ssd_keras#97]
Hi,
First, thank @pierluigiferrari for your detailed and instructive work.
I am new to deep learning stuff.
Is there any tutorial how to detect person or object without ground truth on pre-trained COCO? For example, detecting my dog. I started with your [SSD300 MS COCO Evaluation Tutorial](https://github.com/pierluigiferrari/ssd_keras/blob/master/ssd300_evaluation_COCO.ipynb) and [Weight Sampling Tutorial](https://github.com/pierluigiferrari/ssd_keras/blob/master/weight_sampling_tutorial.ipynb), but I still dont know how to do it.

Thank you.

**Comment by pierluigiferrari** [comment]
That's a question for the [Data Science section of StackOverflow](https://datascience.stackexchange.com/) or for some deep learning forum, it has nothing to do specifically with this SSD implementation.

You could subsample from the MS COCO weights the classes you need (e.g. bicycle, etc.) plus 24 arbitray classes that will eventually learn your the 24 action classes. For example, say you had 10 classes from the MS COCO classes that you were interested in keeping, then you would sub-sample 34 classes, the 10 you want and 24 arbitrary ones. When you sub-sample the classes, you can put them in arbitrary order, so you if you want the classes to be in a specific order, put them in that order. Next, you could create a joint dataset from your action dataset and part of the MS COCO dataset. The JSON parser in `DataGenerator` provides the option to include only certain classes in the dataset, so you just include the classes you need, but you would have to modify the parser slightly so that it throws away any images that don't contain any of the classes you need, otherwise the dataset will include all the original COCO images but with no ground truth object in a lot of them, which you probably wouldn't want. So basically, you split off that part of MS COCO that contains objects of interest for your purpose, and then you merge that part with your action dataset. Then you train the model on the joint dataset.

Note that the fact that the label says "aeroplane" in the image above has nothing to do with the prediction being wrong. What word it says doesn't matter, **you** define that. The model simply predicts class index 1 for the two fencers in the image. The model doesn't care what name you associated with class ID 1. In the trained Pascal VOC weights, class ID 1 is for "aeroplane", and in the trained MS COCO weights, class ID 1 is for "person". You just forgot to change the class names from Pascal VOC to MS COCO.

**Comment by tsly123** [comment]
Super thank @pierluigiferrari ,

Correct me if I am wrong. What you mean is: the network will train on the joint dataset (24 action + object) and it can detect person AND object. Is this included the image in the training set? I mean in training step, can the network detect the object in the image that belongs the action dataset? Or It only works on the test set? My goal is to detect person and object for the whole dataset.

If it only works on the test set, then I will flip the image of the training set and add it into the test set. It should work. At least I think so.
